### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by Zainab-Alzubair**
**In this PR** 

Live Demo: https://aesthetic-cocada-15db6d.netlify.app/

**Config & basic setup**
- Create directories for all Redux state slice files (rockets, missions).

**Redux: Fetch data and update the Redux store**

- Upon first render fetch data from the SpaceX API endpoints:
- Rockets: https://api.spacexdata.com/v3/rockets
- Missions: https://api.spacexdata.com/v3/missions

Once the data are fetched, dispatch an action to store the selected data in the Redux store
**Render UI: lists**

- Render a list of rockets (as per design). For the image of a rocket using the first image in the array of flickr_images.
- Render a table with the missions' data (as per design).

**Redux: Write actions and reducers for booking rockets and joining missions**
**Redux: Write actions and reducers for canceling rockets and leaving missions**
**Render UI: conditional components rendering**

- switch badges and buttons when clicking on the booking/joining button 

**Render UI: My Profile section**

- Render a list of all joined missions (use filter()).
- Render a list of all reserved rockets (use filter()).

**add test for all components and functions.**

Thanks for reviewing our PR

### CODE REVIEW

File: src/__test__/Mission.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        <BrowserRouter>
+          <Missions />
+        </BrowserRouter>
+        ,
+      </Provider>,
+    );
+
+    expect(missions).toMatchSnapshot();
+  });
+});
```
Comment (youmari) [inline src/__test__/Mission.test.js]: - [ ] one of the tests is failing please make sure that tests are passing in order to complete the requirements 
Screenshot 2023-01-05 at 14 26 20

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @FuadNabi @Zainab-Alzubair , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ PRO PR
✔️ No linter error 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/Mission.test.js (hunk 1 above)
Comment (FuadNabi) [reply src/__test__/Mission.test.js]: Thanks for reviewing. Added new snapshot.

**Review by youmari — APPROVED** [review]
Hi @Zainab-Alzubair @FuadNabi , 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 
## Highlights
✔️ PRO PR
✔️ No linter error 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**